python/ray/utils.py

```diff
@@ -29,6 +30,20 @@
 win32_AssignProcessToJobObject = None
 
 
+def get_user_temp_dir():
+    if sys.platform.startswith("darwin") or sys.platform.startswith("linux"):
+        # Ideally we wouldn't need this fallback, but keep it for now for
```

Comment:
The reason for not using `tempfile.gettempdir()` on Mac is that it returns something like `'/var/folders/qf/bnbh0c653814f2kz_p6kjqg40000gn/T'`, which is really difficult for users to navigate to. Instead, we want them to just be able to go to look at `/tmp/ray/session_latest/logs`.

Reply:
Yup I realized, that's why I didn't use that function on Mac :-) eventually we should consider more appropriate directories for different files though, like maybe `${XDG_RUNTIME_DIR}` (fallback `/var/run`) for ephemeral files like sockets, `${XDG_CACHE_HOME}` (with fallback `~/.cache`) for logs, etc., since they can get cleaned up based on different policies, and can help isolate environments from different users, etc.